conda/core/solve.py

```diff
@@ -493,7 +493,9 @@ def _add_specs(self, ssc):
                                          installed_pool):
                     ssc.specs_map[pkg_name] = target_prec.to_match_spec()
                 elif pkg_name in ssc.specs_from_history_map:
-                    ssc.specs_map[pkg_name] = ssc.specs_from_history_map[pkg_name]
+                    ssc.specs_map[pkg_name] = MatchSpec(
+                        ssc.specs_from_history_map[pkg_name],
+                        target=target_prec.dist_str())
```

Comment:
The historic spec is still enforced by the solver, it will not be removed.  The addition of the target [moves the package](https://github.com/conda/conda/blob/4.7.5/conda/resolve.py#L1191-L1204) from the "requested packages" group to the "all other packages" group.  The requested packages group is [optimized early](https://github.com/conda/conda/blob/4.7.5/conda/resolve.py#L1213-L1218) where as all other packages group is [optimized later](https://github.com/conda/conda/blob/4.7.5/conda/resolve.py#L1260-L1269).

With the old code if you can't freeze the historic spec was considered at the same level as the explicit specs, so `conda install anaconda` followed by `conda update ipython` would optimize both the version numbers of `anaconda` and `ipython`.

The current code keeps the historic spec but the explicit specs are given priority.  In the `anaconda` and `ipython` example `conda update ipython` will optimize the version of `ipython` with a constraint that some `anaconda` version must be installed.

This new behavior is in-line with how historic and explicit specs were considered in conda 4.6.